tests/sentry/sentry_metrics/test_cloudspanner_model.py

```diff
@@ -0,0 +1,139 @@
+                    last_seen=datetime.now(),
+                ),
+                SpannerIndexerModel(
+                    id=(uuid.uuid4().int & (1 << 64) - 1) >> 1,
+                    decoded_id=12345,
+                    string="string 2",
+                    organization_id=1,
+                    retention_days=90,
+                    date_added=datetime.now(),
+                    last_seen=datetime.now(),
+                ),
+                SpannerIndexerModel(
+                    id=(uuid.uuid4().int & (1 << 64) - 1) >> 1,
+                    decoded_id=12345,
+                    string="string 3",
+                    organization_id=1,
+                    retention_days=90,
+                    date_added=datetime.now(),
+                    last_seen=datetime.now(),
+                ),
+            ],
+            id="mutation multi write",
+        ),
+        pytest.param(
+            CloudSpannerInsertMode.DML,
+            [
+                SpannerIndexerModel(
+                    id=(uuid.uuid4().int & (1 << 64) - 1) >> 1,
+                    decoded_id=12345,
+                    string="string 11",
+                    organization_id=1,
+                    retention_days=90,
+                    date_added=datetime.now(),
+                    last_seen=datetime.now(),
+                )
+            ],
+            id="dml single write",
+        ),
+        pytest.param(
+            CloudSpannerInsertMode.DML,
+            [
+                SpannerIndexerModel(
+                    id=(uuid.uuid4().int & (1 << 64) - 1) >> 1,
+                    decoded_id=12345,
+                    string="string 12",
+                    organization_id=1,
+                    retention_days=90,
+                    date_added=datetime.now(),
+                    last_seen=datetime.now(),
+                ),
+                SpannerIndexerModel(
+                    id=(uuid.uuid4().int & (1 << 64) - 1) >> 1,
+                    decoded_id=12345,
+                    string="string 13",
+                    organization_id=1,
+                    retention_days=90,
+                    date_added=datetime.now(),
+                    last_seen=datetime.now(),
+                ),
+                SpannerIndexerModel(
+                    id=(uuid.uuid4().int & (1 << 64) - 1) >> 1,
+                    decoded_id=12345,
+                    string="string 14",
+                    organization_id=1,
+                    retention_days=90,
+                    date_added=datetime.now(),
+                    last_seen=datetime.now(),
+                ),
+            ],
+            id="dml multi write",
+        ),
+    ],
+)
+def test_spanner_indexer_write(mode: CloudSpannerInsertMode, models: Sequence[SpannerIndexerModel]):
+    # TODO: Provide instance_id and database_id when running the test
+    spanner_indexer = CloudSpannerIndexer(instance_id="", database_id="")
+    spanner_indexer.validate()
```

Comment:
How do we plan to be able to run this ?

Reply:
We will need to run google cloud spanner emulator. I can plan to work on that once the basic indexer solution has been implemented. For now, I am able to run tests against our shared spanner instances from my local machine.